src/pages/class-list.astro

```diff
@@ -0,0 +1,193 @@
+            data-class-list-target="receiver"
+          >
+            Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolore ea
+            ipsam veritatis hic, beatae expedita, autem, eos voluptatibus
+            possimus fugiat est. Fuga recusandae qui quam animi dolore dolorem
+            quasi illum!
+          </p>
+          <div class="flex space-x-4 mt-5">
+            <Button data-action="class-list#toggle">
+              Toggle classes
+            </Button>
+            <Button data-action="class-list#add">
+              Add classes
+            </Button>
+            <Button data-action="class-list#remove">
+              Remove classes
+            </Button>
+          </div>
+        </div>
+      </div>
+
+      <div slot="code">
+        <Markdown>
+          ```html
+          <div
+            data-controller="class-list"
+            data-class-list-change-class="text-white bg-gradient-to-br from-blue-500 via-fuchsia-500 to-red-500 shadow-xl ring-8 ring-cyan-300"
+          >
+            <p data-class-list-target="receiver">
+              Receiver target
+            </p>
+            <button data-action="class-list#toggle">
+              Toggle classes
+            </button>
+            <button data-action="class-list#add">
+              Add classes
+            </button>
+            <button data-action="class-list#remove">
+              Remove classes
+            </button>
+          </div>
+          ```
+        </Markdown>
+      </div>
+    </CodeDemo>
+
+    <CodeDemo heading="Classes Per Action" className="py-8">
+      <div slot="preview">
+        <div data-controller="class-list">
+          <p
+            class="p-3 border-2 border-white border-opacity-20 rounded-md"
+            data-class-list-target="receiver"
+          >
+            Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolore ea
+            ipsam veritatis hic, beatae expedita, autem, eos voluptatibus
+            possimus fugiat est. Fuga recusandae qui quam animi dolore dolorem
+            quasi illum!
+          </p>
+          <div class="flex space-x-4 mt-5">
+            <Button
+              data-action="class-list#toggle"
+              data-class-list-class-param="bg-blue-500"
+            >
+              Toggle blue bg
+            </Button>
+            <Button
+              data-action="class-list#add"
+              data-class-list-class-param="ring-2 ring-orange-500"
+            >
+              Add orange ring
+            </Button>
+            <Button
+              data-action="class-list#remove"
+              data-class-list-class-param="ring-2 ring-orange-500 bg-blue-500"
+            >
+              Remove orange ring and blue bg
+            </Button>
+          </div>
+        </div>
+      </div>
+
+      <div slot="code">
+        <Markdown>
+          ```html
+          <div data-controller="class-list">
+            <p data-class-list-target="receiver">
```

Comment:
Thought of a potential future enhancement (if it's even a good idea).

The ability to add something like `data-class-list-class-param` to various the `receiver` elements so you could add / remove / toggle different classes on different receivers. 

But maybe there isn't a compelling use case for it.

Reply:
ahh. i see. yeah i was thinking diff receiver behavior would be multiple controllers but that might be too constraining given markup. i think this is a good idea, lemme ponder that

Reply:
Yeah... I think I'm still unsure of the best way to use composition with lots of tiny controllers.

Reply:
noted here https://github.com/vigetlabs/stimulus-controllers/issues/14